Database/FileContentProvider.cs

```diff
@@ -21,10 +22,31 @@ public async Task<IEnumerable<IGenericKarmaItem>> ReadFromFileAsync()
             return list;
         }
 
-        public async Task WriteToFileAsync(IEnumerable<IGenericKarmaItem> listToWrite)
+        public async Task<IEnumerable<IVolunteerUser>> ReadVolunteerFromFileAsync()
+        {
+            List<IVolunteerUser> list = new List<IVolunteerUser>();
+            using (StreamReader reader = new StreamReader(PathToVolunteerFile))
+            {
+                //TODO investigate why await takes so long (might be buffers)
+                string jsonString = reader.ReadToEndAsync().Result;
+                list.AddRange(JsonConvert.DeserializeObject<List<VolunteerUser>>(jsonString));
+            }
+            return list;
+        }
+
+        public async Task WriteItemToFileAsync(IEnumerable<IGenericKarmaItem> listToWrite)
+        {
+            string jsonString = JsonConvert.SerializeObject(listToWrite);
+            using (StreamWriter writer = new StreamWriter(PathToItemFile))
+            {
+                await writer.WriteAsync(jsonString);
+            }
+        }
+
+        public async Task WriteVolunteerToFileAsync(IEnumerable<IVolunteerUser> listToWrite)
         {
             string jsonString = JsonConvert.SerializeObject(listToWrite);
-            using (StreamWriter writer = new StreamWriter(PathToFile))
+            using (StreamWriter writer = new StreamWriter(PathToVolunteerFile))
             {
                 await writer.WriteAsync(jsonString);
```

Comment:
as you can see, `WriteVolunteerToFileAsync` and `WriteItemToFileAsync` are almost the same, so istead we could use one class with generic `IEnumerable<T>` @ArentasMeinorius  do you agree? (same goes for readItemToFileAsync(...) )

Reply:
Masterfully fixed, thank you.